rclcpp/include/rclcpp/executor.hpp

```diff
@@ -107,11 +107,13 @@ class Executor
     }
   }
 
-  void spin_node_once(rclcpp::node::Node::SharedPtr & node, bool nonblocking = false)
+  template<typename T = std::milli>
+  void spin_node_once(rclcpp::node::Node::SharedPtr & node,
+    std::chrono::duration<int64_t, T> timeout = std::chrono::duration<int64_t, T>(-1))
```

Comment:
We might want some typedefs for these durations. And may be a default value for -1.